typed-racket-lib/typed-racket/utils/prefab-c.rkt

```diff
@@ -0,0 +1,248 @@
+#lang racket/base
+
+(require racket/contract)
+
+(provide (rename-out [-prefab/c prefab/c]))
```

Comment:
This should probably have a comment about eagerness of checking and why it's that way (similarity with struct/c). Or maybe it should be more eager. Either way, document in RFC as well.